### DISCUSSION TITLE
RegionsPlugin subscriptions keep increasing after clearing regions

### DISCUSSION THREAD

**Discussion by yannicksengstock** [katspaugh/wavesurfer.js#3942] [Q&A]
Hey there,
I'm using the Regions plugin and have a React app that gets marker positions per WebSocket. These markers can change often. I proceed with a `clearRegions()` call before I add the changed marker positions. Now I noticed two things:

1, That markers still show up on the waveform even though the regions are cleared, when checking with a `getRegions().length` they show 0. (I've seen some issues that could have to do with this reported already)

2. That subscriptions rise up pretty significantly and don't get cleared. (In my case it only takes a few changes to make it rise up to 90000+)

Now I would like to know if I also have to unsubscribe from the past regions manually or if this is a bug?

Thank you in advance!

**Comment by LeonVeganMan** [comment]
Hi all, confirm this. ~500 regions, 5000 subscriptions, then clearRegions (0 regions) but still 5000 subscriptions, add 500 new regions and I've ~10000 subscriptions.

Ciao.
L.

**Reply by katspaugh** [reply]
Thanks for confirming! I've made an [issue](https://github.com/katspaugh/wavesurfer.js/issues/4140) for this.
